File: agent-driven-architecture-main/examples/typescript/src/spine/ports/bus.ts
// ── spine/ports/bus — the append-only timeline (G9, 14.1) ──────────────────
// INTERFACES ONLY (C11).
//
// `append` returns the committed offset. That return value is the ORIGIN of
// every effect key, which is why commit cannot be reordered after perform: the
// key does not exist until the append has returned.

import type { StepIndex } from "../pure/ids";
import type { StepRecord } from "../pure/step-record";

export interface Bus {
  append(record: StepRecord): StepIndex;
  records(): readonly StepRecord[];
}

File: agent-driven-architecture-main/examples/kotlin/spine/src/main/kotlin/adr/spine/ports/Bus.kt
// ── spine/ports/bus — the append-only timeline (G9/14.1) ───────────────────
// append() takes a whole StepRecord and returns the offset it landed at. The
// returned StepIndex is the ORIGIN of every effect key in that step, which is what
// makes "commit strictly precedes perform" a fact about the code rather than a
// convention about the order two lines happen to be written in.

package adr.spine.ports

import adr.spine.pure.StepIndex
import adr.spine.pure.StepRecord

interface Bus {
    fun append(record: StepRecord): StepIndex

    fun records(): List<StepRecord>
}
